authentik/core/models.py

```diff
@@ -887,3 +897,15 @@ def from_request(request: HttpRequest, user: User) -> Optional["AuthenticatedSes
             last_user_agent=request.META.get("HTTP_USER_AGENT", ""),
             expires=request.session.get_expiry_date(),
         )
+
+
+class UserPasswordHistory(models.Model):
```

Comment:
This should probably go alongside the uniquepasswordpolicy model, in its own app (see my comment on that specific model.)

Reply:
I can see both sides; keeping it next to the User model keeps it grouped with User-related concerns, while putting it in with the policy sub-app makes the dependency clear.

I'll wait for @BeryJu to chime in on this one before I move it.

Reply:
True. Another somewhat related instance where we put the model in the app is with stage_consent, where the user consents are in the app instead of core. Core is quite crowded already, which is why I suggested this initially. Let's wait for what Jens has to say

Reply:
Ended up moving UserPasswordHistory into `policies.unique_password` as you suggested :+1: